commons-numbers-examples/examples-jmh/src/main/java/org/apache/commons/numbers/examples/jmh/arrays/EuclideanNormEvaluator.java

```diff
@@ -0,0 +1,248 @@
+         * @return ulp error mean
+         */
+        public double getUlpErrorMean() {
+            return ulpErrorMean;
+        }
+
+        /** Get the ulp error standard deviation.
+         * @return ulp error standard deviation
+         */
+        public double getUlpErrorStdDev() {
+            return ulpErrorStdDev;
+        }
+
+        /** Get the ulp error minimum value.
+         * @return ulp error minimum value
+         */
+        public double getUlpErrorMin() {
+            return ulpErrorMin;
+        }
+
+        /** Get the ulp error maximum value.
+         * @return ulp error maximum value
+         */
+        public double getUlpErrorMax() {
+            return ulpErrorMax;
+        }
+
+        /** Get the number of failed computations, meaning the number of
+         * computations that overflowed or underflowed.
+         * @return number of failed computations
+         */
+        public int getFailCount() {
+            return failCount;
+        }
+    }
+
+    /** Class used to accumulate statistics during a norm evaluation run.
+     */
+    private static final class StatsAccumulator {
+
+        /** Sample index. */
+        private int sampleIdx;
+
+        /** Array of ulp errors for each sample. */
+        private final double[] ulpErrors;
+
+        /** Construct a new instance.
+         * @param count number of samples to be accumulated
+         */
+        StatsAccumulator(final int count) {
+            ulpErrors = new double[count];
+        }
+
+        /** Report a computation result.
+         * @param expected expected result
+         * @param actual actual result
+         */
+        public void report(final double expected, final double actual) {
+            ulpErrors[sampleIdx++] = Double.isFinite(actual) && actual != 0.0 ?
+                    computeUlpDifference(expected, actual) :
+                    Double.NaN;
+        }
+
+        /** Compute the final statistics for the run.
+         * @return statistics object
+         */
+        public Stats computeStats() {
+            int successCount = 0;
+            double sum = 0d;
+            double min = Double.POSITIVE_INFINITY;
+            double max = Double.NEGATIVE_INFINITY;
+
+            for (double ulpError : ulpErrors) {
+                if (Double.isFinite(ulpError)) {
+                    ++successCount;
+                    min = Math.min(ulpError, min);
+                    max = Math.max(ulpError, max);
+                    sum += ulpError;
+                }
+            }
+
+            final double mean = sum / successCount;
+
+            double diffSumSq = 0d;
+            double diff;
+            for (double ulpError : ulpErrors) {
+                if (Double.isFinite(ulpError)) {
+                    diff = ulpError - mean;
+                    diffSumSq += diff * diff;
+                }
+            }
+
+            final double stdDev = Math.sqrt(diffSumSq / (successCount - 1));
```

Comment:
In the ridiculous case where success count is 1 this will be invalid. zero successes will compute a NaN mean which is a useful marker, and stdDev will be zero.